Fix mautic#14245 Mautic doesn't show error when choosing remote provider

PedroLoureiro1 opened the pull request:
Discovered that it was necessary to validate client_id and client_secret configurations before proceeding. If they are missing, an exception is thrown and properly handled. Additionally, I implemented specific error messages to inform users about failures when loading files from the selected remote storage provider.

<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 7.x)
* a.b for any bug fixes (e.g. 5.2, 6.0)
* c.x for any bug fixes, features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 7.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✔️ <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | ❌
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ❌ <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #14245  <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

Mautic doesn't provide error when selecting unavailable provider Discovered that it was necessary to validate client_id and client_secret configurations before proceeding. If they are missing, an exception is thrown and properly handled. Additionally, I implemented specific error messages to inform users about failures when loading files from the selected remote storage provider

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
<!--
| Before                                 | After
| -------------------------------------- | ---
|                                        | 
-->


---
### 📋 Steps to test this PR:

1.Go to Plugins
2.Click the button Install plugins
3.Go to Components > Assets
4.Create a new asset
5.In Storage location, select Remote
6.Click Browse Remote Files
7.Click Amazon S3
8.See the PHP error message (dev env)
<!--
This part is crucial. Take the time to write very clear, annotated and step by step test instructions, because testers may not be developers.
-->


<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

PedroLoureiro1 committed:
Fix mautic#14245 Mautic doesn't show error when choosing remote provider

Discovered that it was necessary to validate client_id and client_secret
configurations before proceeding. If they are missing, an exception is
thrown and properly handled. Additionally, I implemented specific error
messages to inform users about failures when loading files from the
selected remote storage provider.

escopecz commented on plugins/MauticCloudStorageBundle/EventListener/RemoteAssetBrowseSubscriber.php:
```diff
@@ -24,6 +24,10 @@ public function onAssetRemoteBrowse(Events\RemoteAssetBrowseEvent $event): void
-        $event->setAdapter($integration->getAdapter());
+        try {
+            $event->setAdapter($integration->getAdapter());
+        } catch (\Exception $e) {
```
Let's throw and catch a specific exception. This could mask some other exception that we'd like to record in the logs with the stack trace.

PedroLoureiro1 replied:
Hi @escopecz,

I’ve made changes to the `RemoteAssetBrowseSubscriber.php` to handle exceptions more specifically. Previously, a general exception was caught, which could potentially mask other important exceptions. I’ve now replaced this with the specific `InvalidCredentialConfigurationException` to ensure that other exceptions can still be logged properly with their stack traces.

Could you please review the changes and let me know if everything looks good? If you have any suggestions or concerns, feel free to share them.

Thanks in advance for your time!

Best regards,  
Pedro Loureiro

escopecz reviewed (changes requested):
I was able to replicate the issue:
![Screenshot 2025-04-15 at 16 13 46](https://github.com/user-attachments/assets/a1fe529e-6744-4921-8d95-b8eec28f1882)
And this PR fixes it:
![Screenshot 2025-04-15 at 16 16 18](https://github.com/user-attachments/assets/640511a1-621e-49fc-9060-39baac8c86c0)
I have one suggestion for the code changes.

PedroLoureiro1 committed:
Fix mautic#14245 Mautic doesn't show error when choosing remote provider
- Replaced general exception handling with the specific
InvalidCredentialConfiguration Exception.

escopecz commented on app/bundles/AssetBundle/Assets/js/asset.js:
```diff
@@ -28,7 +28,8 @@ Mautic.updateRemoteBrowser = function(provider, path) {
                 mQuery('.remote-file-search').quicksearch('#remoteFileBrowser .remote-file-list a');
             } else {
-                // TODO - Add error handler
+                const flashMessage = Mautic.addErrorFlashMessage(Mautic.translate('mautic.asset.update.remote.browser.error'));
+                Mautic.setFlashes(flashMessage);
```
Can we use the error message from the flash message generated by the server instead of doing this in JS?

escopecz commented on app/bundles/CoreBundle/Translations/en_US/javascript.ini:
```diff
@@ -18,4 +18,5 @@ mautic.core.tabs.more="more"
-mautic.user.config.title.experience_and_learning.reset_confirmation="All dismissed resources visible again"
\ No newline at end of file
+mautic.user.config.title.experience_and_learning.reset_confirmation="All dismissed resources visible again"
+mautic.asset.update.remote.browser.error="Service selection failed. Please verify that your credentials are correctly configured."
```
Because to separate the concerns, the CoreBundle shouldn't contain code specific for the AssetBundle.

PedroLoureiro1 replied:
Hi @escopecz,

With this update, the error message now uses Mautic.addErrorFlashMessage together with Mautic.setFlashes, ensuring that the message is styled consistently with the rest of the application and properly dismissible by the user. Additionally, the response message from the backend is reused directly, without hardcoded fallback or translation strings.

Could you please review the changes and let me know if they align with the current UI standards and error handling practices in Mautic? I'm happy to make any adjustments you suggest.

Thanks in advance!

Best regards,
Pedro Loureiro

escopecz reviewed (approved):
That's pretty much perfect!

PedroLoureiro1 committed:
Fix mautic#14245 Remote provider error not displayed to user

- Replaced generic HTML flash creation with Mautic.setFlashes
+ addErrorFlashMessage.
- Uses response.message directly without translation for
accurate error context.

escopecz reviewed (approved):
Works like a charm. I see no more issues in the code 👍 
<img width="939" alt="Screenshot 2025-04-29 at 12 54 05" src="https://github.com/user-attachments/assets/cf42b999-0cd9-4146-ad15-196934ed43dd" />

escopecz committed:
Merge branch '7.x' into fix-UX-Mautic-doesnt-provide-error-message-when-selecting-unavailable-remote-storage-provider-#14245

escopecz commented:
@all-contributors please add @PedroLoureiro1 for code
